### DISCUSSION TITLE
Plugins

### DISCUSSION THREAD

**Discussion by fugkco** [firefly-iii/firefly-iii#5009] [General]
Hi James. I just want to bring back the discussion on plugins. It was previously raised in #1954 and was rejected.

I understand where you're coming from, however, I'd like to discuss additional things. I understand wanting to limit the scope features, however, Firefly III has come a long way since that discussions, and has gotten a lot more features since the previous discussion. Firely III has become a bit of a monolith that contains all the features with no way of disabling them whether or not they're used.

I understand security is a concern, as it should be. However, just like Mozilla and Chrome, that is a concern for the user. Especially considering Firefly is self-hosted _only_, it requires individuals to be a bit more savvy than the average person. It is up to them to figure out not install bad plugins.

I'd love it if you could re-consider adding plugins.

**Comment by JC5** [comment]
Hey, thanks for the discussion. I'm going to blunt about this. There is never going to be space for Firefly III plug-ins. No zips to download, no marketplace, nothing. 

It's not so much about plug-ins being unsafe but plug-ins *becoming* unsafe. This is what's happening to Wordpress all the time. At this point, Wordpress is basically a backdoor with a blogging platform attached to it. Security is indeed the users responsibility, an excellent argument not to do it.

Rather, I want to continue developing the API, and make *that* the plug-in space people are looking for. If Firefly III is a platform, extra features and options can be built on top of the API. To make progress here, I am building the new Firefly III layout entirely on top of the API. So anything Firefly III can do is in the API, and more if you feel like building it.

The various importers are already plug-ins in a sense, they use the API and nothing else. 

Insofar as Firefly III is a monolith, I respectfully disagree. There are plenty of features but Firefly III is pretty basic compared to Mint, YNAB and GNUcash.
